rocks-db/src/asset.rs

```diff
@@ -3,38 +3,1106 @@ use std::collections::HashMap;
+            dynamic_details: None,
+            authority: None,
+            owner: None,
+            collection: None,
+        }
+    }
+}
+
+impl AssetCompleteDetails {
+    pub fn convert_to_fb_bytes(&self) -> Vec<u8> {
+        let mut builder = FlatBufferBuilder::new();
+        let asset_complete_details = self.convert_to_fb(&mut builder);
+        builder.finish_minimal(asset_complete_details);
+        builder.finished_data().to_vec()
+    }
+
+    pub fn convert_to_fb<'a>(
+        &self,
+        builder: &mut FlatBufferBuilder<'a>,
+    ) -> WIPOffset<fb::AssetCompleteDetails<'a>> {
+        let pk = Some(builder.create_vector(&self.pubkey.to_bytes()));
+        let static_details = self
+            .static_details
+            .as_ref()
+            .map(|sd| asset_static_details_to_fb(builder, sd));
+        let dynamic_details = self
+            .dynamic_details
+            .as_ref()
+            .map(|dd| asset_dynamic_details_to_fb(builder, dd));
+        let authority = self
+            .authority
+            .as_ref()
+            .map(|a| asset_authority_to_fb(builder, a));
+        let owner = self.owner.as_ref().map(|o| asset_owner_to_fb(builder, o));
+        let collection = self
+            .collection
+            .as_ref()
+            .map(|c| asset_collection_to_fb(builder, c));
+        fb::AssetCompleteDetails::create(
+            builder,
+            &fb::AssetCompleteDetailsArgs {
+                pubkey: pk,
+                static_details,
+                dynamic_details,
+                authority,
+                owner,
+                collection,
+            },
+        )
+    }
+}
+
+impl<'a> From<fb::AssetCompleteDetails<'a>> for AssetCompleteDetails {
+    fn from(value: fb::AssetCompleteDetails<'a>) -> Self {
+        let pubkey = Pubkey::try_from(value.pubkey().unwrap().bytes()).unwrap();
+        AssetCompleteDetails {
+            pubkey,
+            static_details: value.static_details().map(From::from),
+            dynamic_details: value.dynamic_details().map(From::from),
+            authority: value.authority().map(From::from),
+            owner: value.owner().map(From::from),
+            collection: value.collection().map(From::from),
+        }
+    }
+}
+
+impl<'a> From<fb::AssetStaticDetails<'a>> for AssetStaticDetails {
+    fn from(value: fb::AssetStaticDetails<'a>) -> Self {
+        let pubkey = Pubkey::try_from(value.pubkey().unwrap().bytes()).unwrap();
+        let edition_address = value
+            .edition_address()
+            .map(|ea| Pubkey::try_from(ea.bytes()).unwrap());
+        AssetStaticDetails {
+            pubkey,
+            specification_asset_class: value.specification_asset_class().into(),
+            royalty_target_type: value.royalty_target_type().into(),
+            created_at: value.created_at(),
+            edition_address,
+        }
+    }
+}
+
+impl<'a> From<fb::AssetDynamicDetails<'a>> for AssetDynamicDetails {
```

Comment:
At first glance, it seems that if `fb::AssetDynamicDetails` has any `None` field then conversion fails. Doesn't it mean it shouldn't contain any `None` in the first place?